presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1431,8 +1431,8 @@ private RelationType analyzeView(Query query, QualifiedObjectName name, Optional
                     .setTransactionId(session.getTransactionId().orElse(null))
                     .setIdentity(identity)
                     .setSource(session.getSource().orElse(null))
-                    .setCatalog(catalog.orElse(null))
-                    .setSchema(schema.orElse(null))
+                    .setCatalog(catalog.orElse(session.getCatalog().orElse(null)))
+                    .setSchema(schema.orElse(session.getSchema().orElse(null)))
```

Comment:
Actually, I think this is wrong.  The catalog and view should only be the one recorded in the view definition